source/isaaclab/isaaclab/envs/direct_marl_env.py

```diff
@@ -126,6 +130,17 @@ def __init__(self, cfg: DirectMARLEnvCfg, render_mode: str | None = None, **kwar
         else:
             self.viewport_camera_controller = None
 
+        # create event manager
+        # note: this is moved here to allow USD-related randomization events that need to be
+        #   applied before the simulation starts.
```

Comment:
```suggestion
        # note: this is needed here (rather than after simulation play) to allow USD-related randomization events
        #   that must happen before the simulation starts. Example: randomizing mesh scale
```